src/NzbDrone.Core/Download/Clients/Flood/Flood.cs

```diff
@@ -0,0 +1,221 @@
+
+            return hash;
+        }
+
+        protected override string AddFromMagnetLink(RemoteEpisode remoteEpisode, string hash, string magnetLink)
+        {
+            _proxy.AddTorrentByUrl(magnetLink, HandleTags(remoteEpisode, Settings), Settings);
+
+            return hash;
+        }
+
+        public override IEnumerable<DownloadClientItem> GetItems()
+        {
+            var items = new List<DownloadClientItem>();
+
+            var list = _proxy.GetTorrents(Settings);
+
+            foreach (var torrent in list)
+            {
+                var properties = torrent.Value;
+
+                if (!Settings.Tags.All(tag => properties.Tags.Contains(tag)))
+                {
+                    continue;
+                }
+
+                var item = new DownloadClientItem
+                {
+                    DownloadClientInfo = DownloadClientItemClientInfo.FromDownloadClient(this),
+                    DownloadId = torrent.Key,
+                    Title = properties.Name,
+                    OutputPath = _remotePathMappingService.RemapRemoteToLocal(Settings.Host, new OsPath(properties.Directory)),
+                    Category = properties.Tags.Count > 0 ? properties.Tags[0] : null,
+                    RemainingSize = properties.SizeBytes - properties.BytesDone,
+                    RemainingTime = TimeSpan.FromSeconds(properties.Eta),
+                    TotalSize = properties.SizeBytes,
+                    SeedRatio = properties.Ratio,
+                    Message = properties.Message,
+                };
+
+                if (properties.Status.Contains("error"))
+                {
+                    item.Status = DownloadItemStatus.Warning;
+                }
+                else if (properties.Status.Contains("seeding") || properties.Status.Contains("complete"))
+                {
+                    item.Status = DownloadItemStatus.Completed;
+                }
+                else if (properties.Status.Contains("downloading"))
+                {
+                    item.Status = DownloadItemStatus.Downloading;
+                }
+                else if (properties.Status.Contains("stopped"))
+                {
+                    item.Status = DownloadItemStatus.Paused;
+                }
+
+                if (Settings.TargetRatio != -1 && item.SeedRatio > Settings.TargetRatio)
+                {
+                    item.CanMoveFiles = item.CanBeRemoved = item.Status == DownloadItemStatus.Completed;
+                }
+                else
+                {
+                    item.CanMoveFiles = item.CanBeRemoved = false;
+                }
+
+                items.Add(item);
+            }
+
+            return items;
+        }
+
+        public override DownloadClientItem GetImportItem(DownloadClientItem item, DownloadClientItem previousImportAttempt)
+        {
+            var result = item.Clone();
+
+            var contentPaths = _proxy.GetTorrentContentPaths(item.DownloadId, Settings);
```

Comment:
Unfortunately, no. Get torrent API call provides directory path only. There are many cases: single-file vs. multi-file, base directory vs. sub directory.

When the torrent is single-file, Sonarr expects path to the file. Otherwise, it will try to import the whole directory, which can cause problems if there are unrelated files in the directory. For multi-file torrents, Sonarr expects path to the base directory of the torrent.

As such, an additional API call to fetch the list of contents of a torrent is required. It is more efficient (resource saving) to fetch when the import is requested.